### PR TITLE
Fix query failure if a table deletion involves deviceId with null & Fix status code when deleting illegal time range & support is null in deletion

### PR DESCRIPTION
**Pull request by jt2594838**

### CODE REVIEW

**Review by JackieTien97 — APPROVED** [review]

File: integration-test/src/test/java/org/apache/iotdb/relational/it/db/it/IoTDBDeletionTableIT.java [hunk 1]
```diff
@@ -570,6 +571,52 @@ public void testMultiDevice() throws SQLException {
...
+        Statement statement = connection.createStatement()) {
+      statement.execute("use test");
+      statement.execute(
+          "create table t" + testNum + " (id1 string id, id2 string id, s1 int32 measurement)");
+      // id1 is null for this record
+      statement.execute("insert into t" + testNum + " (time, id2, s1) values (1, '1', 1)");
+      statement.execute("insert into t" + testNum + " (time, id2, s1) values (2, '', 2)");
+      statement.execute("flush");
+
+      statement.execute("delete from t" + testNum + " where id1 = NULL and time <= 1");
```
Comment (JackieTien97) [inline integration-test/src/test/java/org/apache/iotdb/relational/it/db/it/IoTDBDeletionTableIT.java:587]: should be id1 IS NULL

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/AnalyzeUtils.java [hunk 2]
```diff
@@ -540,13 +547,14 @@ private static IDPredicate getIdPredicate(
     String rightHandValue;
     if (right instanceof StringLiteral) {
       rightHandValue = ((StringLiteral) right).getValue();
+    } else if (right instanceof NullLiteral) {
+      rightHandValue = null;
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/AnalyzeUtils.java:551]: result of XXX = NULL is NULL, so in such case, we should never delete any rows.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: integration-test/src/test/java/org/apache/iotdb/relational/it/db/it/IoTDBDeletionTableIT.java (hunk 1 above)
Comment (jt2594838) [reply integration-test/src/test/java/org/apache/iotdb/relational/it/db/it/IoTDBDeletionTableIT.java]: Done

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/AnalyzeUtils.java (hunk 2 above)
Comment (jt2594838) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/AnalyzeUtils.java]: Done

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**